src/storage-client/src/types/connections.rs

```diff
@@ -165,6 +168,35 @@ pub enum Connection {
     AwsPrivatelink(AwsPrivatelinkConnection),
 }
 
+impl Connection {
+    /// Whether this connection should be validated by default on creation.
+    pub fn validate_by_default(&self) -> bool {
+        match self {
+            Connection::Kafka(conn) => conn.validate_by_default(),
+            Connection::Csr(conn) => conn.validate_by_default(),
+            Connection::Postgres(conn) => conn.validate_by_default(),
+            Connection::Ssh(conn) => conn.validate_by_default(),
+            Connection::Aws(conn) => conn.validate_by_default(),
+            Connection::AwsPrivatelink(conn) => conn.validate_by_default(),
```

Comment:
Some connections cannot be validated on creation because there is additional setup steps to be done that happen after the connection is created. For example creating an SSH connection requires getting the mz public keys and putting them in your bastion but you only get access to them once the connection is created. At the same time we want to have validation by default for the connections that are expected to work in the common cases.